qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Cleanup.java

```diff
@@ -0,0 +1,120 @@
+import java.util.List;
+
+import static org.elasticsearch.packaging.util.FileUtils.getTempDir;
+import static org.elasticsearch.packaging.util.FileUtils.lsGlob;
+import static org.elasticsearch.packaging.util.Platforms.isAptGet;
+import static org.elasticsearch.packaging.util.Platforms.isDPKG;
+import static org.elasticsearch.packaging.util.Platforms.isRPM;
+import static org.elasticsearch.packaging.util.Platforms.isSystemd;
+import static org.elasticsearch.packaging.util.Platforms.isYUM;
+
+public class Cleanup {
+
+    private static final List<String> ELASTICSEARCH_FILES_LINUX = Arrays.asList(
+        "/usr/share/elasticsearch",
+        "/etc/elasticsearch",
+        "/var/lib/elasticsearch",
+        "/var/log/elasticsearch",
+        "/etc/default/elasticsearch",
+        "/etc/sysconfig/elasticsearch",
+        "/var/run/elasticsearch",
+        "/usr/share/doc/elasticsearch",
+        "/usr/lib/systemd/system/elasticsearch.conf",
+        "/usr/lib/tmpfiles.d/elasticsearch.conf",
+        "/usr/lib/sysctl.d/elasticsearch.conf"
+    );
+
+    // todo
+    private static final List<String> ELASTICSEARCH_FILES_WINDOWS = Collections.emptyList();
+
+    public static void cleanEverything() {
+        final Shell sh = new Shell();
+
+        // kill elasticsearch processes
+        if (Platforms.WINDOWS) {
+
+            // the view of processes returned by Get-Process doesn't expose command line arguments, so we use WMI here
+            sh.run("powershell.exe", "-Command",
+                "Get-WmiObject Win32_Process | " +
+                "Where-Object { $_.CommandLine -Match 'org.elasticsearch.bootstrap.Elasticsearch' } | " +
+                "ForEach-Object { $_.Terminate() }");
+
+        } else {
+
+            sh.run("pkill", "-u", "elasticsearch");
+            sh.run("bash", "-c", "ps aux | grep -i 'org.elasticsearch.bootstrap.Elasticsearch' | awk {'print $2'} | xargs kill -9");
+
+        }
+
+        if (Platforms.LINUX) {
+            purgePackagesLinux();
+        }
+
+        // remove elasticsearch users
+        if (Platforms.LINUX) {
+            sh.run("userdel", "elasticsearch");
+            sh.run("groupdel", "elasticsearch");
+        }
+
+        // delete files that may still exist
+        lsGlob(getTempDir(), "elasticsearch*").forEach(FileUtils::rm);
+        final List<String> filesToDelete = Platforms.WINDOWS
+            ? ELASTICSEARCH_FILES_WINDOWS
+            : ELASTICSEARCH_FILES_LINUX;
+        filesToDelete.stream()
+            .map(Paths::get)
+            .filter(Files::exists)
+            .forEach(FileUtils::rm);
+
+        // disable elasticsearch service
+        // todo add this for windows when adding tests for service intallation
+        if (Platforms.LINUX && isSystemd()) {
+            sh.runAndAssertSuccess("systemctl", "unmask", "systemd-sysctl.service");
+        }
+    }
+
+    private static void purgePackagesLinux() {
+        final Shell sh = new Shell();
+
+        if (isRPM()) {
```

Comment:
Aren't these all mutually exclusive?

Reply:
They should be (at least deb vs rpm). That said I think the logic here does make sense (if any of these package managers exist on the system, remove the packages from them). It's the same [in the bats tests](https://github.com/elastic/elasticsearch/blob/a8cea90e1084983b2d045edae84187fe8708399d/qa/vagrant/src/test/resources/packaging/utils/utils.bash#L289-L307)